internal/server/singleprocess/service_job.go

```diff
@@ -380,15 +378,14 @@ func (s *service) onDemandRunnerStartJob(
 		return nil, "", status.Errorf(codes.Aborted, "error configuring expiration: %s", err)
 	}
 
-	// Our start runner job can go anywhere.
-	// NOTE(mitchellh): For more targetted runners in multi-cluster mode,
-	// we'll probably want this to be controllable by runner profiles
-	// so that start tasks get sent to the right cluster.
-	job.TargetRunner = &pb.Ref_Runner{
-		Target: &pb.Ref_Runner_Any{
-			Any: &pb.Ref_RunnerAny{},
-		},
+	profile, err := s.state.OnDemandRunnerConfigGet(source.OndemandRunner)
+	if err != nil {
+		return nil, "", status.Errorf(codes.FailedPrecondition,
+			"Failed to get on-demand runner config by name %q, id %q",
+			job.OndemandRunner.Name, job.OndemandRunner.Id)
```

Comment:
```suggestion
			"Failed to get on-demand runner config by name %q, id %q: %s",
			job.OndemandRunner.Name, job.OndemandRunner.Id, err)
```

Could include the error message here for completeness, it should add a little bit more extra context, just in case.